include/fixed_containers/fixed_doubly_linked_list_storage.hpp

```diff
@@ -0,0 +1,114 @@
+      : IMPLEMENTATION_DETAIL_DO_NOT_USE_storage_{}
+      , IMPLEMENTATION_DETAIL_DO_NOT_USE_chain_{}
+    {
+        // the last element in the chain array is the start/end sentinel, so must start by pointing
+        // to itself
+        next_of(MAXIMUM_SIZE) = MAXIMUM_SIZE;
+        prev_of(MAXIMUM_SIZE) = MAXIMUM_SIZE;
+    }
+
+public:
+    [[nodiscard]] constexpr bool full() const noexcept { return storage().full(); }
+
+    constexpr const T& at(const IndexType i) const { return storage().at(i); }
+    constexpr T& at(const IndexType i) { return storage().at(i); }
+
+    [[nodiscard]] constexpr IndexType front_index() const { return next_of(MAXIMUM_SIZE); }
+    [[nodiscard]] constexpr IndexType back_index() const { return prev_of(MAXIMUM_SIZE); }
+
+    template <typename... Args>
+    constexpr IndexType emplace_after_index_and_return_index(IndexType idx, Args&&... args)
+    {
+        IndexType new_idx = storage().emplace_and_return_index(std::forward<Args>(args)...);
+        next_of(new_idx) = next_of(idx);
+        prev_of(next_of(new_idx)) = new_idx;
+        prev_of(new_idx) = idx;
+        next_of(idx) = new_idx;
+
+        return new_idx;
+    }
+    template <typename... Args>
+    constexpr IndexType emplace_before_index_and_return_index(IndexType idx, Args&&... args)
+    {
+        return emplace_after_index_and_return_index(prev_of(idx), std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    constexpr IndexType emplace_back_and_return_index(Args&&... args)
+    {
+        return emplace_after_index_and_return_index(back_index(), std::forward<Args>(args)...);
+    }
+    template <typename... Args>
+    constexpr IndexType emplace_front_and_return_index(Args&&... args)
+    {
+        return emplace_before_index_and_return_index(front_index(), std::forward<Args>(args)...);
+    }
+
+    constexpr IndexType delete_at_and_return_next_index(IndexType idx)
+    {
+        storage().delete_at_and_return_repositioned_index(idx);
+        next_of(prev_of(idx)) = next_of(idx);
+        prev_of(next_of(idx)) = prev_of(idx);
+
+        return next_of(idx);
+    }
+
+public:
+    [[nodiscard]] constexpr const IndexType& next_of(IndexType i) const
+    {
+        return chain().at(i).next;
+    }
+    [[nodiscard]] constexpr IndexType& next_of(IndexType i) { return chain().at(i).next; }
+
+    [[nodiscard]] constexpr const IndexType& prev_of(IndexType i) const
+    {
+        return chain().at(i).prev;
+    }
+    [[nodiscard]] constexpr IndexType& prev_of(IndexType i) { return chain().at(i).prev; }
+
+private:
+    constexpr const StorageType& storage() const
+    {
+        return IMPLEMENTATION_DETAIL_DO_NOT_USE_storage_;
+    }
+    constexpr StorageType& storage() { return IMPLEMENTATION_DETAIL_DO_NOT_USE_storage_; }
+
+    constexpr const ChainType& chain() const { return IMPLEMENTATION_DETAIL_DO_NOT_USE_chain_; }
+    constexpr ChainType& chain() { return IMPLEMENTATION_DETAIL_DO_NOT_USE_chain_; }
+};
```

Comment:
How does this work without a destructor? Don't we need to call the destructor of the populated slots of `FixedIndexPoolBasedStorage` if the underlying type isn't `triviallyDestructible`?
(this question also extends to copying and moving)

Reply:
Will put memory management semantics to this.